src/transformStateWithClones.js

```diff
@@ -1,13 +1,44 @@
 'use strict';
 
+function addProperties(target, source) {
+  return Object.assign(target, source);
+}
+
+function removeProperties(obj, properties) {
+  properties.forEach(key => {
+    delete obj[key];
+  });
+}
+
 /**
  * @param {Object} state
  * @param {Object[]} actions
  *
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const history = [];
+  let object = { ...state };
```

Comment:
object is a bad name, it can be `currentState` or something similar that describes the variable